crowdsource/resolvers/person.py

```diff
@@ -0,0 +1,91 @@
+def apply_person_patches(jur_name):
+    count = 0
+    patches = IssueResolverPatch.objects.filter(jurisdiction__name=jur_name,
+                                                status='approved')
+    image_duplicates = []
+    name_duplicates = []
+    for patch in patches:
+        person = Person.objects.get(id=patch.object_id)
+        issue = None
+        if patch.category == 'image':
+            # Number of approved pathces for 'image' for a person must be one.
+            ap = patches.filter(object_id=person.id, category='image').count()
+            if ap == 1:
+                if person.image == patch.old_value:
+                    # make sure that patch is not applied before.
+                    if not person.image == patch.new_value:
+                        person.image = patch.new_value
+                        person.save()
+                        count += 1
+                else:
+                    if patch.old_value:
+                        issue = 'wrong-photo'
+                    else:
+                        issue = 'person-missing-photo'
+
+            else:
+                if patch.object_id not in image_duplicates:
+                    p = Person.objects.get(id=patch.object_id).name
+                    print("{}: Found {} `approved` patches of `image` "
+                          "for \"{}\". skipping..."
+                          .format(jur_name, ap, p))
+                image_duplicates.append(patch.object_id)
+        elif patch.category == 'name':
+            # Number of approved pathces for 'name' for a person must be one.
+            ap = patches.filter(object_id=person.id, category='name').count()
+            if ap == 1:
+                if person.name == patch.old_value:
+                    # make sure that patch is not applied before.
+                    if not person.name == patch.new_value:
+                        person.name = patch.new_value
+                        person.save()
+                        count += 1
+                else:
+                    issue = 'wrong-name'
+            else:
+                if patch.object_id not in name_duplicates:
+                    p = Person.objects.get(id=patch.object_id).name
+                    print("{}: Found {} `approved` patches of `name` "
+                          "for \"{}\". skipping..."
+                          .format(jur_name, ap, p))
+                name_duplicates.append(patch.object_id)
+        elif patch.category in ['voice', 'address', 'email']:
+            contact = person.contact_details.filter(type=patch.category,
+                                                    value=patch.old_value)
+            if contact:
+                if not person.contact_details.filter(type=patch.category,
+                                                     value=patch.new_value):
+                    contact.update(value=patch.new_value)
```

Comment:
this will set all of their phone numbers to the same one!

Reply:
I am not sure, I am following, "test_wrong_voice_patch_over_ride_scraper", this test is passing, and it does something similar, as one approved scraper is getting applied and other one, gets over ridden and both are getting tested later. Agreed, .update may not be good use case here, since it does update all at once, but I am checking for old_value here.